tools/webpack-plugin/src/BacktraceWebpackSourceGenerator.ts

```diff
@@ -0,0 +1,39 @@
+import { ContentAppender, DebugIdGenerator } from '@backtrace/sourcemap-tools';
+import type { Source } from 'webpack-sources';
+import { ConcatSource, RawSource, SourceMapSource } from 'webpack-sources';
+
+export class BacktraceWebpackSourceGenerator {
+    constructor(
+        private readonly _debugIdGenerator: DebugIdGenerator,
+        private readonly _contentAppender: ContentAppender,
+    ) {}
+
+    public addDebugIdToSource(source: Source, debugId: string): ConcatSource {
+        const sourceSnippet = this._debugIdGenerator.generateSourceSnippet(debugId);
+        return new ConcatSource(source, sourceSnippet);
```

Comment:
source snippet is a string, how it will behave with the current behavior of concatSource?

Reply:
`ConcatSource` accepts `string` as any parameter.